build/scripts/k8s-export-openapi/main.go

```diff
@@ -0,0 +1,470 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main implements a program to convert from export-openapi.sh to Go script.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/itchyny/json2yaml"
+)
+
+func main() {
+	tmpDir := "../../tmp"
+
+	if _, err := os.Stat(tmpDir); err == nil {
+		err = os.RemoveAll(tmpDir)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	err := os.MkdirAll(tmpDir, os.ModePerm)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Start the kubectl proxy.
+	cmd := exec.Command("kubectl", "proxy")
+	err = cmd.Start()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Make an HTTP request to fetch the OpenAPI JSON.
+	resp, err := http.Get("http://127.0.0.1:8001/openapi/v2")
```

Comment:
We may want to throw a few seconds of Sleep above this, just to make sure the proxy is ready before we query it.

Reply:
Included -> 5 seconds sleep time, to stop the proxy if already is running at the beginning, and graceful termination at the end.